### ISSUE TITLE
FEATURE: allow LAN access by default, even using `pnpm run start`

### ISSUE THREAD

**Issue by PieBru** [coleam00/bolt.new-any-llm#41]
**Is your feature request related to a problem? Please describe:**
I tried this fork also on a Proxmox VE 8.2 LXC Arch Linux container.
It works, except it doesn't allow connections from LAN clients, only `localhost`.
I suppose it may be the old problem of serving on `localhost` instead of serving on `0.0.0.0`.

**Describe the solution you'd like:**
Allow LAN clients to connect.

**Describe alternatives you've considered:**
ssh tunnelling, but it may be complex for the average user.

**Additional context:**

**Comment by goncaloalves** [comment]
This is fairly straightforward:

in the vite.config.ts file, inside the defineConfig add:
`server: { host: '0.0.0.0'},},`

Restart the app (pnpm run dev)

And you should see it being served on your LAN IP.

**Comment by wonderwhy-er** [comment]
In one of my projects I use  https://github.com/localtunnel/localtunnel
It's easy to add

But I am not sure what you are really trying to do.

Oh I guess you want to run it on your Linux server and expose it so other machines on LAN or over the internet can use it?

I did achieve hat with a hack but it's a bad idea.

Thing is that currently there are 3 ways to use it

npm run dev - that runs live dev server that is not a good idea for production

npm run build + npm run dev - that also runs a dev variant

and npm run build + npm run deploy - that deploys it to cloudflare workers


I have not have luck yet to run it locally or on Linux server in production mode
It looks like one would need to use this
https://github.com/cloudflare/workerd

Basically, this project is written for CloudFlare worker environment and does not really run natively on Node/Deno/Bun 

Only two ways forward are:
1. Rewrite it for Node.js instead of Cloudflare workers which would put it in conflict with upstream version from StackBlitz
2. I am trying but failing to use that workerd runtime to run it instead of node.js
3. Try to allow building for both workerd and node.js - best hardest option

Anyone else interested?

**Comment by PieBru** [comment]
> server: { host: '0.0.0.0'},},

I am not able to find a place where to put this patch in `./vite.cinfig.ts` nor `./node_modules/.pnpm/vite-plugin-node-polyfills@0.22.0_rollup@4.18.0_vite@5.3.1_@types+node@20.14.9_sass@1.77.6_/node_modules/vite-plugin-node-polyfills/shims/vite.config.ts`

**Comment by PieBru** [comment]
> In one of my projects I use https://github.com/localtunnel/localtunnel

Nice! Is it secure in your experience?

> Oh I guess you want to run it on your Linux server and expose it so other machines on LAN or over the internet can use it?

Yes, I like the LocalLlama lifestyle https://www.reddit.com/r/LocalLLaMA/

**Comment by goncaloalves** [comment]
you have to add it yourself. 

Here is how I did it:

![image](https://github.com/user-attachments/assets/c51d686f-a4dd-4da8-994c-ae0b3c080761)

**Comment by PieBru** [comment]
@goncaloalves Thank you, but I tryed with:
```typescript
...
export default defineConfig((config) => {
  return {
    server: { 
      host: '0.0.0.0',
    },
    build: {
      target: 'esnext',
    },
...
```

It builds OK, but doesn't solve.
Port 8788 remains still closed and inaccessible from the LAN.

**Comment by goncaloalves** [comment]
In my machine it defaults to port 5173 as all vite apps do.

Did you try that port? where are you getting/ configuring por 8788?

![image](https://github.com/user-attachments/assets/6dcf3462-1565-4c75-a32d-74f178d5eb46)

**Comment by PieBru** [comment]
@goncaloalves  Good catch!

I did not set any port, it's all at default values.
`pnpm run dev` works on port 5173, exposing it (listen on 0.0.0.0).
`pnpm run start` works locally on port 8788, but it doesn't expose it (listen on 127.0.0.1)

![image](https://github.com/user-attachments/assets/c29be299-931a-4038-a516-e6364df90fc6)

![image](https://github.com/user-attachments/assets/6fa4ed16-53c1-40be-9cc2-bf0273a0bf4b)

Thank you for your patience.
Piero

**Comment by PieBru** [comment]
Just as a reminder, now a Bolt LXC local "dev" server works only if Bolt is installed on the same LXC together with Ollama, at least until #43 is solved.

**Comment by PieBru** [comment]
It works!

![image](https://github.com/user-attachments/assets/6a824c1e-41cc-4f18-906c-c110ec20a7a7)

The exact line is:
`"start": "bindings=$(./bindings.sh) && wrangler pages dev ./build/client $bindings --port 5173 --ip 0.0.0.0",`

**Comment by goncaloalves** [comment]
I think this issue is not a feature and, instead, a configuration. 

I suggest updating the README.md with the instructions available here.

**Comment by PieBru** [comment]
I agree 100%.
